### PR TITLE
Improve generated video poster file name

### LINKED ISSUE #3228: Improve generated video poster file name
Noticed this while looking into #1218.

In `useUploadVideoFrame` we retrieve the first frame of a video (for the poster image) and then upload it to WordPress via `useUploader` / `uploadFile` and ultimately the API provider's `uploadMedia`.

Whenever we do that, the uploaded poster image gets named by WordPress like so: `image-1`, `image-2`, `image-3`, etc. - regardless of the original video's title or file name.

This is because of this line in `uploadMedia`:

```js
file.name || file.type.replace('/', '.') // turns image/png into image
```

I think we should improve this to allow setting better image names, and the key is the `uploadFile` callback.

Idea:

First, add a second `data` parameter to `uploadFile`. Then do something like `const additionalData = { post: storyId, media_source: 'editor', ...data }` in `uploadFile`. This allows passing additional data when uploading the file.

In `useUploadVideoFrame` we can now pass a second argument to `uploadFile` to set a custom `title` à la "Generated Video Poster" or something.

Finally, `uploadMedia` can take this `title` and derive a file name from there, but still fall back to `file.name` and `file.type`.

**Edit:** Ideally, the poster name would just be the video name + suffix. So if the video in the WordPress media library  is named `awesome-video.mp4`, the poster could be named `awesome-video-poster.png`

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add a better file name to poster. 

## Relevant Technical Choices

This change is simple, take the video's file name and add -image.jpeg to it. So `amazing-video.mp4` becomes `amazing-video-poster.jpeg`. 

## Testing Instructions

---

Fixes #3228

### CODE REVIEW

**Comment by swissspidy** [comment]
Is it actually a JPEG, not a PNG?

Edit: nvm, it's a JPEG:

https://github.com/google/web-stories-wp/blob/8d5a62489044fe9f483a9bfbe6fb826d37408497/assets/src/edit-story/app/media/utils/getFirstFrameOfVideo.js#L61

File: assets/src/edit-story/app/media/utils/useUploadVideoFrame.js [hunk 1]
```diff
@@ -97,6 +98,15 @@ function useUploadVideoFrame({ updateMediaElement }) {
     }
   };
 
+  /**
+   * Helper function get the file name from url.
+   *
+   * @param url
+   * @return {string}
+   */
+  const getFileName = (url) =>
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/utils/useUploadVideoFrame.js:107]: There's already `getFileName.js` from `edit-story/utils` that could perhaps be used.

It doesn't have any tests (or actual usage 🤔 ) though, so not sure it works in this context.

File: assets/src/edit-story/app/media/utils/useUploadVideoFrame.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ function useUploadVideoFrame({ updateMediaElement }) {
   const processData = async (id, src) => {
     try {
       const obj = await getFirstFrameOfVideo(src);
+      obj.name = getFileName(src) + '-poster.jpeg';
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/utils/useUploadVideoFrame.js:49]: Neat! So simple!

File: assets/src/edit-story/app/media/utils/useUploadVideoFrame.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/media/utils/useUploadVideoFrame.js:107]: I tried using it. But it returned with the ext. So `filename.mp4`. This function just gives the filename without the ext.

**Comment by swissspidy** [comment]
@spacedmonkey FYI there's a lint error

**Comment by spacedmonkey** [comment]
@swissspidy Can you review again?

**Review by swissspidy — APPROVED** [review]

**Pull request merged**